modules/dasBidAdapter.js

```diff
@@ -0,0 +1,240 @@
+  return params.get('npa') == '1';
+}
+
+function buildOpenRTBRequest(bidRequests, bidderRequest) {
+  const { customParams, keyValues } = parseParams(
+    bidRequests[0].params,
+    bidderRequest,
+  );
+  const imp = bidRequests.map((bid, index) => {
+    const sizes = getAdUnitSizes(bid);
+    const imp = {
+      id: bid.bidId,
+      tagid: bid.params.slot,
+      secure: 1,
+    };
+
+    if (bid.mediaTypes?.banner) {
+      imp.banner = {
+        format: sizes.map((size) => ({
+          w: size[0],
+          h: size[1],
+        })),
+        pos: bid.params.slotSequence || 0,
+      };
+    }
+
+    return imp;
+  });
+
+  const request = {
+    id: bidderRequest.bidderRequestId,
+    imp,
+    site: {
+      id: customParams.site,
+      page: customParams.du,
+      ref: customParams.dr,
+      ext: {
+        area: customParams.area,
+      },
+      ...bidderRequest.ortb2.site,
+    },
+    device: bidderRequest.device,
+    user: {
+      ext: {
+        ids: buildUserIds(customParams),
+      },
+    },
+    ext: {
+      network: customParams.network,
+      key_values: keyValues,
+    },
+    at: 1,
+    tmax: bidderRequest.timeout,
+  };
+
+  if (bidderRequest.gdprConsent) {
+    request.user = {
+      consent: bidderRequest.gdprConsent.consentString,
+      ext: {
+        gdpr: bidderRequest.gdprConsent.gdprApplies,
+        dsa:
+          customParams.dsainfo !== undefined
+            ? { required: customParams.dsainfo }
+            : undefined,
+        npa: getNpaFromPubConsent(customParams.pubconsent),
+        ...request.user.ext,
+      },
+    };
+  }
+
+  return request;
+}
+
+function interpretResponse(serverResponse) {
+  const bidResponses = [];
+  const response = serverResponse.body;
+
+  if (!response || !response.seatbid || !response.seatbid.length) {
+    return bidResponses;
+  }
+
+  response.seatbid.forEach((seatbid) => {
+    seatbid.bid.forEach((bid) => {
+      const bidResponse = {
+        requestId: bid.impid,
+        cpm: bid.price,
+        currency: response.cur || 'USD',
+        width: bid.w,
+        height: bid.h,
+        creativeId: bid.crid || bid.id,
+        netRevenue: true,
+        ttl: 300,
+        meta: {
+          advertiserDomains: bid.adomain || [],
+        },
+      };
+      // first implementation only supports banner
+      if (bid.mtype === 1) {
+        bidResponse.mediaType = BANNER;
+        bidResponse.ad = bid.adm;
+        bidResponses.push(bidResponse);
+      }
+    });
+  });
+
+  return bidResponses;
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  isBidRequestValid: function (bid) {
+    if (!bid || !bid.params) {
+      return false;
+    }
+    return !!(bid.params?.network && bid.params?.site && bid.params?.area && bid.params?.slot);
+  },
+
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const data = buildOpenRTBRequest(validBidRequests, bidderRequest);
+    return {
+      method: 'POST',
```

Comment:
Nie popłyniemy z kosztami CloudFronta?

Reply:
Na ten moment się tym nie przejmujemy, jeśli będzie to problem będziemy szukać jakiejś innej drogi. OpenRTB można użyć tylko wykorzystując metodę typu POST